hnn_core/tests/test_network.py

```diff
@@ -258,6 +251,38 @@ def test_network():
     assert len(net.connectivity) == 0
 
 
+def test_add_cell_type():
+    """Test adding a new cell type."""
+    from copy import deepcopy
+
+    params = read_params(params_fname)
+    net = default_network(params)
+
+    n_total_cells = net.n_cells
+    pos = [(0, idx, 0) for idx in range(10)]
+    tau1 = 0.6
+
+    new_cell = deepcopy(net.cell_types['L2_basket'])
+    net._add_cell_type('new_type', pos=pos, cell_template=new_cell)
+    net.cell_types['new_type'].p_syn['ampa']['tau1'] = tau1
+
+    n_new_type = len(net.gid_ranges['new_type'])
+    assert n_new_type == len(pos)
+    # xxx: why can't I do L2_basket -> new_type ?
+    # no key for network_builder.ncs
```

Comment:
@ntolley any idea what's going on here?

Reply:
never mind, I figured this out!

Reply:
Ah sorry I missed this! Yeah the internal code for defining where synapses are made require an association between cell parts and `distal`, `proximal`, etc.